linker/symbols.go

```diff
@@ -602,15 +600,69 @@ func (s *Symbols) AddExtensionDeclaration(extension, extendee protoreflect.FullN
 			// This is a declaration that has already been added. Ignore.
 			return nil
 		}
-		return handler.HandleErrorf(span, "extension %s already declared as extending %s with tag %d at %v", extension, existing.extendee, existing.tag, existing.pos)
+		return handler.HandleErrorf(span, "extension %s already declared as extending %s with tag %d at %v", extension, existing.extendee, existing.tag, existing.span.Start())
 	}
 	if s.extDecls == nil {
 		s.extDecls = map[protoreflect.FullName]extDecl{}
 	}
 	s.extDecls[extension] = extDecl{
-		pos:      span.Start(),
+		span:     span,
 		extendee: extendee,
 		tag:      tag,
 	}
 	return nil
 }
+
+// Lookup finds the registered location of the given name. If the given name has
+// not been seen/registered, nil is returned.
+func (s *Symbols) Lookup(name protoreflect.FullName) ast.SourceSpan {
+	pkg := name.Parent()
+	for {
+		pkgSyms := s.getPackage(pkg, false)
```

Comment:
`getPackage` with `exact = false` looks like it can't return nil. Is this for loop needed?

Reply:
Indeed, good catch!